samza-api/src/main/java/org/apache/samza/application/StreamApplicationDescriptor.java

```diff
@@ -41,16 +42,19 @@
    * <p>
    * If the default system descriptor is set, it must be set <b>before</b> creating any input/output/intermediate streams.
    * <p>
-   * If an input/output stream is created with a stream-level Serde, they will be used, else the serde specified
-   * for the {@code job.default.system} in configuration will be used.
+   * A {@code defaultSystemSerde} may be provided for intermediate streams created on the default system. If the
+   * intermediate stream is explicitly created with a stream-level Serde, it will be used. Else, if a
+   * non-null {@code defaultSystemSerde} has been provided, it will be used. Else the key and msg serde specified
+   * in configuration for the {@code job.default.system} will be used.
    * <p>
    * Providing an incompatible message type for the intermediate streams that use the default serde will result in
    * {@link ClassCastException}s at runtime.
    *
    * @param defaultSystemDescriptor the default system descriptor to use
+   * @param defaultSystemSerde the {@link Serde} to use for streams on the default system
```

Comment:
The commit description states "Require providing serde", but it looks like it is also ok to pass null here. Just wanted to double check that it is ok to pass null. If it is ok to pass null, could you please note in the param doc that this param is nullable?